### ISSUE TITLE
Basic authentication with Python API does not work in `main` Airflow

### ISSUE THREAD

**Issue by potiuk** [apache/airflow#36377]
### Apache Airflow version

main (development)

### What happened?

Looks like API authentication does not work with `main` version of airflow in `breeze`.  The API calls are refused with 401 HTTP  authentication error.

### What you think should happen instead?

The client should authenticate with basic_auth.

### How to reproduce

When you test Python client with https://github.com/apache/airflow-client-python/blob/main/dev/test_python_client.py (and earlier configure the webserver with:

```
auth_backends = airflow.api.auth.backend.session,airflow.api.auth.backend.basic_auth
```

the webserver refuses the connection with:

```
[2023-12-22T22:06:59.292+0000] {security_handler_factory.py:382} INFO - ... No auth provided. Aborting with 401.
                                                                                                                                                  │127.0.0.1 - admin [22/Dec/2023:22:06:59 +0000] "GET /api/v1/config HTTP/1.1" 401 124 "-" "OpenAPI-Generator/2.7.3/python"
```

This is the same for `apache-airflow-client==2.7.3` and `apache-airflow-client==2.8.0` . 

When you use `--use-airflow-version==2.8.0`, it works just fine, so the problem seems to be coming from some recent changes in main.

### Operating System

Breeze CI image (debian-bookworm)

### Versions of Apache Airflow Providers

main version

### Deployment

Other

### Deployment details

Breeze

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by potiuk** [comment]
cc: @vincbeck

**Comment by vincbeck** [comment]
Interesting ... When I run Airflow using Breeze (`breeze start-airflow --skip-asset-compilation --backend postgres`), I can successfully get the list of DAGs by calling the API `http://localhost:28080/api/v1/dags` in Postman using Basic auth

**Comment by potiuk** [comment]
Hmm. let me try again. Maybe that was something I made mistake about (though I did it a the time I verified Python client and this is where I found it).

**Comment by potiuk** [comment]
Yep. Can't reproduce. It works just fine in main. Must have been some mistake I've done (or something was fixed since in main :D ).

Sorry for the noise.
